src/Writer/PhpArray.php

```diff
@@ -139,4 +158,54 @@ protected function processIndented(array $config, array $arraySyntax, &$indentLe
 
         return $arrayString;
     }
+
+    /**
+     * Process a string configuration value
+     *
+     * @param string $value
+     * @return string
+     */
+    protected function processStringValue($value)
+    {
+        if ($this->useClassNameScalars && $this->checkStringIsFqn($value)) {
+            return $value . '::class';
+        }
+
+        return var_export($value, true);
+    }
+
+    /**
+     * Process a string configuration key
+     *
+     * @param string $key
+     * @return string
+     */
+    protected function processStringKey($key)
+    {
+        if ($this->useClassNameScalars && $this->checkStringIsFqn($key)) {
+            return $key . '::class';
+        }
+
+        return "'" . addslashes($key) . "'";
+    }
+
+    /**
+     * Check whether a string represents a resolvable FQCN
+     *
+     * @param string $string
+     * @return bool
+     */
+    protected function checkStringIsFqn($string)
+    {
+        if (strlen($string) < 1) {
+            return false;
+        }
+
+        if ($string[0] !== "\x5C") {
+            // Prepend a backslash, ensuring we check against a FQN.
+            $string = "\x5C" . $string;
+        }
+
+        return class_exists($string) || interface_exists($string);
```

Comment:
Indeed. If autoloading fails, its run against everytime on interface_* and trait_*
I'v tried it another way:

```php
spl_autoload_call($string);

return class_exists($string, false) || interface_exists($string, false) || trait_exists($string, false);
```

This passes.